### DISCUSSION TITLE
Caddy Home Assistant OS/Supervised modalità DNS Challenge

### DISCUSSION THREAD

**Discussion by sempreinsieme** [indomus/forum#2311] [HUB: Home Assistant]
Ciao a tutti, qualcuno mi aiuta a trovare una guida Caddy in modalità DNS Challenge. Le porte 443 e 80 sono già impegnate in un altra applicazione su un altro pc, quindi, non saprei come configurare caddy se non in DNS Challenge. Se avete suggerimenti sono ben accetti. Grazie!!
PS. se possibile configurandolo con no-ip.

**Comment by snakuzzo** [comment]
Mi pare non ci sia la versione precompilata di Caddy2 con i moduli per le DNS Challenge...o almeno non c'è quella con i moduli duckdns, cloudflare e mi sa nemmeno no-ip.

Io non uso l'addon perché di base uso Core, ma per test ho creato un Dockerfile semplice semplice dove ho inserito...

```
FROM caddy:builder AS builder

RUN xcaddy build --with github.com/caddy-dns/cloudflare --with github.com/caddy-dns/duckdns

FROM caddy:latest

COPY --from=builder /usr/bin/caddy /usr/bin/caddy
```

in questo modo viene fatta la build del container con i moduli per le DNS challenge di cloudflare e duckdns, però parto direttamente dall'immagine `caddy:builder`.
Se guardi il [Dockerfile dell'addon in questione](https://github.com/einschmidt/addon-caddy-2/blob/main/caddy-2/Dockerfile) invece, fa il download della versione precompilata di Caddy (a seconda dell'architettura su cui deve girare).

Detto ciò...
A prescindere da tutto il modulo per no-ip non esiste, quindi dovresti cominciare con il seguire [questi step](https://caddy.community/t/how-to-use-dns-provider-modules-in-caddy-2/8148#if-you-do-not-find-your-dns-provider-4) per capire come implementare quello che ti serve, per poi riportarlo eventualmente in una tua fork del progetto dell'addon...o magari per fare una PR dell'addon stesso.

EDIT: Dimenticavo...per fare la DNS Challenge con noi-ip devi avere account a pagamento perché devi poter scrivere il record TXT.
Ma se vai su no-ip.com vedrai questo...

![image](https://user-images.githubusercontent.com/7205371/215791403-6c946822-8105-4f4b-9578-c839f27612e3.png)

Quindi hai perso a prescindere, a meno che tu non abbia voglia di pagare.

**Reply by sempreinsieme** [reply]
> Mi pare non ci sia la versione precompilata di Caddy2 con i moduli per le DNS Challenge...o almeno non c'è quella con i moduli duckdns, cloudflare e mi sa nemmeno no-ip.
> 
> Io non uso l'addon perché di base uso Core, ma per test ho creato un Dockerfile semplice semplice dove ho inserito...
> 
> ```
> FROM caddy:builder AS builder
> 
> RUN xcaddy build --with github.com/caddy-dns/cloudflare --with github.com/caddy-dns/duckdns
> 
> FROM caddy:latest
> 
> COPY --from=builder /usr/bin/caddy /usr/bin/caddy
> ```
> 
> in questo modo viene fatta la build del container con i moduli per le DNS challenge di cloudflare e duckdns, però parto direttamente dall'immagine `caddy:builder`. Se guardi invece il [Dockerfile dell'addon in questione](https://github.com/einschmidt/addon-caddy-2/blob/main/caddy-2/Dockerfile) invece, fa il download della versione precompilata di Caddy (a seconda dell'architettura su cui deve girare).
> 
> Detto ciò... A prescindere da tutto il modulo per no-ip non esiste, quindi dovresti cominciare con il seguire [questi step](https://caddy.community/t/how-to-use-dns-provider-modules-in-caddy-2/8148#if-you-do-not-find-your-dns-provider-4) per capire come implementare quello che ti serve, per poi riportarlo eventualmente in una tua fork del progetto dell'addon...o magari per fare una PR dell'addon stesso.
> 
> EDIT: Dimenticavo...per fare la DNS Challenge con noi-ip devi avere account a pagamento perché devi poter scrivere il record TXT. Ma se vai su no-ip.com vedrai questo...
> 
> ![image](https://user-images.githubusercontent.com/7205371/215791403-6c946822-8105-4f4b-9578-c839f27612e3.png)
> 
> Quindi hai perso a prescindere, a meno che tu non abbia voglia di pagare.

Ok, grazie per l risposta, vediamo se ho capito. NO-IP lo scartiamo a priori in quanto è a pagamento :) .  Quindi, devo creare una dokker per candy in modo da poter utilizzare duckdns? Scusa ma non conosco molto caddy e non so se può servirmi, in quanto, cercavo semplicemente un modo per non usare le porte 443 e 80 che uso per nextcloud su un altra macchina.

**Reply by snakuzzo** [reply]
No...il succo del discorso è che se sei su HA OS e vuoi la DNS Challenge con questo addon, devi lasciar perdere questa soluzione.

**Reply by sempreinsieme** [reply]
> No...il succo del discorso è che se sei su HA OS e vuoi la DNS Challenge con questo addon, devi lasciar perdere questa soluzione.

Ottimo... :) hai qualche suggerimento per ovviare il problema? A me è venuto in mente solo di creare una VPN, quindi, lo raggiungo facilmente ma non era l'obbiettivo principale. Grazie in anticipo.

**Reply by snakuzzo** [reply]
Cambi le porte con cui espone le altre applicazioni e lasci la 443 o la 80 per le rispettive tls-alpn-01 e http-01 challenge

**Reply by sempreinsieme** [reply]
> Cambi le porte con cui espone le altre applicazioni e lasci la 443 o la 80 per le rispettive tls-alpn-01 e http-01 challenge

grazie, sei stato gentilissimo, quasi quasi mi conviene installare home assistant come dokker sull'altro pc cosi viaggia in parallelo a nextcloud.
